Lib/importlib/__init__.py

```diff
@@ -85,7 +85,18 @@ def import_module(name, package=None):
             if character != '.':
                 break
             level += 1
-    return _bootstrap._gcd_import(name[level:], package, level)
+    module = _bootstrap._gcd_import(name[level:], package, level)
+    if module:
+        sys.audit(
```

Comment:
Should not this be emitted before an attempt to import the module?

Also, I think it should not ignore `level` and should emit an absolute name. It is better to do this in `_gcd_import()` to avoid a double work. Please add a test for relative import.

Reply:
Agreed it should be before - we usually audit after basic parameter validation (i.e. is it the expected type) and before any operations begin.

Because we're reusing the `import` event, try to get the behaviour as close as possible to the other places it is used (so that code already handling it will still work). I suspect that means we want `name` to be as absolute as possible, and `path` is best effort (so skipping it if we aren't sure of it is fine - `sys.path` is passed for anyone who cares enough to figure it out).

I haven't figured out how to do that here, so if Serhiy is right, then Serhiy is right :-)

Reply:
Good feedback, I moved the location of the call.